cluster/replication/shard_replication_apply.go

```diff
@@ -265,6 +265,35 @@ func (s *ShardReplicationFSM) DeleteReplicationsByTenants(collection string, ten
 	return nil
 }
 
+func (s *ShardReplicationFSM) HasOngoingReplication(collection string, shard string, replica string) bool {
+	s.opsLock.RLock()
+	defer s.opsLock.RUnlock()
+
+	sourceFQDN := newShardFQDN(replica, collection, shard)
+	ops, ok := s.opsBySourceFQDN[sourceFQDN]
+	if !ok {
+		return false
+	}
+
+	ongoingOp := false
+	for _, op := range ops {
+		status, ok := s.statusById[op.ID]
+		if !ok {
+			return false
+		}
+		switch status.GetCurrentState() {
+		case api.CANCELLED:
```

Comment:
is there "cleanup" going on during the cancelled state that would interact with de/activation? i don't remember how long we expect the cancelled state to last but if it's not too slow, perhaps we could consider cancelled ops as having ongoing replication too. let me know your thoughts, you are more familiar with the cancelled stuff that me